pkg/cmd/admin/image/verify-signature.go

```diff
@@ -0,0 +1,368 @@
+package image
+
+import (
+	"errors"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"strings"
+
+	"github.com/containers/image/docker/policyconfiguration"
+	"github.com/containers/image/docker/reference"
+	"github.com/containers/image/signature"
+	sigtypes "github.com/containers/image/types"
+	"github.com/openshift/origin/pkg/client"
+	"github.com/openshift/origin/pkg/cmd/templates"
+	"github.com/openshift/origin/pkg/cmd/util/clientcmd"
+	imageapi "github.com/openshift/origin/pkg/image/api"
+
+	"github.com/spf13/cobra"
+
+	kapi "k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/api/unversioned"
+	kcmdutil "k8s.io/kubernetes/pkg/kubectl/cmd/util"
+)
+
+var (
+	verifyImageSignatureLongDesc = templates.LongDesc(`
+	Verifies the imported image signature using the local public key.
+
+	This command verifies if the signature attached to an image is trusted by
+	using the provided public GPG key.
+	Trusted image means that the image signature was signed by a valid GPG key and the image identity
+	provided by the signature content matches with the image.
+	By default, this command will not record a signature condition back to the Image object but only
+	print the verification status to the console.
+
+	To record a new condition, you have to pass the "--confirm" flag.
+
+	Note that verifying single signature using different "--public-key" or "--expected-identity"
+	will clear verification status from all the other image signatures not signed by this
+	key.
+	`)
+
+	verifyImageSignatureExample = templates.Examples(`
+	# Verify the image signature using the local GNUPG keychan and record the status as a condition to image
+	%[1]s sha256:c841e9b64e4579bd56c794bdd7c36e1c257110fd2404bebbb8b613e4935228c4 --expected-identity=registry.local:5000/foo/bar:v1
+	`)
+)
+
+type VerifyImageSignatureOptions struct {
+	InputImage        string
+	ExpectedIdentity  string
+	PublicKeyFilename string
+	PublicKey         []byte
+	Confirm           bool
+	Remove            bool
+	CurrentUser       string
+
+	Client client.Interface
+	Out    io.Writer
+	ErrOut io.Writer
+}
+
+func NewCmdVerifyImageSignature(name, fullName string, f *clientcmd.Factory, out, errOut io.Writer) *cobra.Command {
+	opts := &VerifyImageSignatureOptions{ErrOut: errOut, Out: out}
+	cmd := &cobra.Command{
+		Use:     fmt.Sprintf("%s IMAGE EXPECTED_IDENTITY [--confirm]", name),
+		Short:   "Verify that the given IMAGE signature is trusted",
+		Long:    verifyImageSignatureLongDesc,
+		Example: fmt.Sprintf(verifyImageSignatureExample, name),
+		Run: func(cmd *cobra.Command, args []string) {
+			kcmdutil.CheckErr(opts.Complete(f, cmd, args, out))
+			kcmdutil.CheckErr(opts.Run())
+		},
+	}
+
+	cmd.Flags().BoolVar(&opts.Confirm, "confirm", opts.Confirm, "If true, the result of the verification will be recorded to an image object.")
```

Comment:
`--record=false: Record current kubectl command in the resource annotation.`

This is clearely not this case, since `--record` will do more than recording an annotation. We also have `--confirm` for pruning and importing image stream, so I will want to keep it `--confirm` just for the sake of consistency.... you spend too much time in kubectl world @soltysh :P